### ISSUE TITLE
BaseRankingEvaluator scores the wrong entities when the evaluation pool is a subset of classes

### ISSUE THREAD

**Issue by ferzcam** [bio-ontology-research-group/mowl#146]
## Summary

`BaseRankingEvaluator.get_expanded_scores` scores the wrong entities whenever the evaluation pool is a subset of the ontology's classes. It substitutes **pool positions** (`0..n-1`) for the entity column, but the evaluation model reads that column as **global entity ids**. The ranking is therefore computed against arbitrary classes and the resulting metrics are noise.

This affects `GDAEvaluator`, `PPIEvaluator` and `SubsumptionEvaluator`, since all three inherit `RankingEvaluator`, which builds its pools from `class_to_id` — i.e. global ids of a subset.

It fails silently: no exception, and the metrics look plausible.

## Where

`mowl/evaluation/base.py`, `get_expanded_scores` (~line 99):

```python
eval_tails = th.arange(len(self.tails), device=self.device).repeat(num_batch_heads).unsqueeze(1)
...
eval_heads = th.arange(len(self.heads), device=self.device).repeat(num_batch_tails).unsqueeze(1)
```

## Why it is wrong

Two id spaces are in play:

* **global entity ids** — `0..num_classes-1` over every class in the ontology
* **pool positions** — `0..n-1` over the evaluation pool only

`RankingEvaluator.__init__` builds the pools in the *global* space:

```python
evaluation_heads_tensor = th.tensor([self.class_to_id[c] for c in eval_heads.as_str], dtype=th.long)
```

and the evaluation model consumes the *global* space, e.g. `EvaluationModel` in
`mowl/models/graph_kge/graph_pykeen_model.py`:

```python
def forward(self, data, *args, **kwargs):
    logits = self.kge_model.score_hrt(data)     # pykeen: columns are global entity ids
    return - logits
```

`get_expanded_scores` emits pool positions into that column, so the two spaces are mixed.

The class already knows the difference: `__init__` builds `self.mapped_heads` / `self.mapped_tails` for exactly this conversion, and `update_filtering_labels` uses them correctly. Only the scoring path skips them.

The bug is invisible when the pool happens to be the first `n` classes, because then `arange(n) == self.tails` and the two spaces coincide.

## Minimal example

Ontology with 8 classes:

```
id  entity
 0  HP:0001250   (phenotype)
 1  HP:0002376   (phenotype)
 2  MP:0003631   (phenotype)
 3  HP:0000118   (phenotype)
 4  BRCA1        (gene)
 5  TP53         (gene)
 6  Disease_A
 7  Disease_B
```

Ranking diseases for a gene gives `heads = [4, 5]`, `tails = [6, 7]`.

Expected scoring:

```
score(4, associated_with, 6)     BRCA1 -> Disease_A
score(4, associated_with, 7)     BRCA1 -> Disease_B
```

Actual scoring, because `arange(2) == [0, 1]`:

```
score(4, associated_with, 0)     BRCA1 -> HP:0001250
score(4, associated_with, 1)     BRCA1 -> HP:0002376
```

The diseases are never scored.

## Evidence

Same trained TransD model, same test edges, ranking 1,003 held-out gene–disease pairs over a pool of 4,936 diseases inside a 175,861-class ontology. The only difference between the two columns is the entity ids used for the candidates:

| metric | `arange` (current) | `self.tails` | mowl-borg 0.3.0 |
|---|---|---|---|
| `f_hits@1` | 0.0 | 0.23529411764705882 | 0.23529411764705882 |
| `f_hits@10` | 0.0 | 0.5044865403788634 | 0.5044865403788634 |
| `f_mr` | 2527.58 | 121.71585244267199 | 121.71585244267199 |
| `auc` | 0.4876 | 0.9750 | 0.9759 |

With `arange`, AUC 0.4876 on 4,936 candidates is chance. With `self.tails`, hits and mean rank match mowl-borg 0.3.0's `EmbeddingsRankBasedEvaluator` to the last digit on the same model.

(The small AUC gap, 0.9750 vs 0.9759, is a separate and harmless convention difference: 0.3.0 decrements `worst_rank` for skipped tails, whereas 2.x passes `num_tails` into `compute_rank_roc`.)

## Suggested fix

```python
eval_tails = self.tails.repeat(num_batch_heads).unsqueeze(1)
eval_heads = self.heads.repeat(num_batch_tails).unsqueeze(1)
```

Nothing downstream needs to change. `head_scores.view(-1, len(self.tails))` still makes column *j* correspond to `self.tails[j]`, `compute_ranking_metrics` already converts the true entity to a pool position via `th.where(self.tails == tail)`, and `filtering_labels` remains indexed by position. The expansion pattern must stay as it is, heads with `repeat_interleave(len(tails))` and tails with `repeat(n_heads)`, so the flattened pairs stay row-major and match the `view`.

## Suggested regression test

Evaluate over a deliberately **non-contiguous** pool (e.g. entity ids `[6, 7]` in an 8-class ontology, or any pool not starting at 0) and assert the AUC is above chance for a model known to fit the data. A test whose pool is the full class set cannot catch this, since both id spaces coincide there.

## Smaller, separate issues noticed nearby

Happy to file these individually if useful:

1. `BaseRankingEvaluator.__init__` (base.py:52) compares the caller's `heads`/`tails` against a device-resident `arange` rather than the copies it just moved to the device, so CPU tensors with `device="cuda"` raise `Expected all tensors to be on the same device`.
2. Filtering is multiplicative (`preds * 10000`) and only demotes a pair when scores are non-negative; with raw scores that can be negative, a filtered pair sorts to rank 1 instead. `compute_ranking_metrics` has no `activation` argument, unlike 0.3.0 which applied one explicitly.
3. Ties are broken by `argsort` position rather than averaged as `scipy.rankdata` does.
4. `mowl/evaluation/rank_based.py` cannot be constructed at all in 2.x: `RankBasedEvaluator` calls `super().__init__(device)` positionally while the rewritten `Evaluator.__init__` takes `dataset` first, and `embeddings_to_dict` no longer exists.
